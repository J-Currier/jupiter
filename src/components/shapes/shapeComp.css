
.fade-out {
	-webkit-animation: fade-out 0.5s ease-out both;
	        animation: fade-out 0.5s ease-out both;
}

 @-webkit-keyframes fade-out {
    0% {
      opacity: 1;
    }
    100% {
      opacity: 0.8;
    }
  }
  @keyframes fade-out {
    0% {
      opacity: 0.8;
    }
    100% {
      opacity: 0;
    }
  }

  @-webkit-keyframes fade-in {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
  @keyframes fade-in {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }

  .fade-in {
	-webkit-animation: fade-in 0.5s cubic-bezier(0.390, 0.575, 0.565, 1.000) both;
	        animation: fade-in 0.5s cubic-bezier(0.390, 0.575, 0.565, 1.000) both;
}

.shake-horizontal {
	-webkit-animation: shake-horizontal 0.5s cubic-bezier(0.455, 0.030, 0.515, 0.955) both;
	        animation: shake-horizontal 0.5s cubic-bezier(0.455, 0.030, 0.515, 0.955) both;
}
@-webkit-keyframes shake-horizontal {
  0%,
  100% {
    -webkit-transform: translateX(0);
            transform: translateX(0);
  }
  10%,
  30%,
  50%,
  70% {
    -webkit-transform: translateX(-10px);
            transform: translateX(-10px);
  }
  20%,
  40%,
  60% {
    -webkit-transform: translateX(10px);
            transform: translateX(10px);
  }
  80% {
    -webkit-transform: translateX(8px);
            transform: translateX(8px);
  }
  90% {
    -webkit-transform: translateX(-8px);
            transform: translateX(-8px);
  }
}
@keyframes shake-horizontal {
  0%,
  100% {
    -webkit-transform: translateX(0);
            transform: translateX(0);
  }
  10%,
  30%,
  50%,
  70% {
    -webkit-transform: translateX(-10px);
            transform: translateX(-10px);
  }
  20%,
  40%,
  60% {
    -webkit-transform: translateX(10px);
            transform: translateX(10px);
  }
  80% {
    -webkit-transform: translateX(8px);
            transform: translateX(8px);
  }
  90% {
    -webkit-transform: translateX(-8px);
            transform: translateX(-8px);
  }
}

.shake-vertical {
	-webkit-animation: shake-vertical 0.5s cubic-bezier(0.455, 0.030, 0.515, 0.955) both;
	        animation: shake-vertical 0.5s cubic-bezier(0.455, 0.030, 0.515, 0.955) both;
}

@-webkit-keyframes shake-vertical {
  0%,
  100% {
    -webkit-transform: translateY(0);
            transform: translateY(0);
  }
  10%,
  30%,
  50%,
  70% {
    -webkit-transform: translateY(-8px);
            transform: translateY(-8px);
  }
  20%,
  40%,
  60% {
    -webkit-transform: translateY(8px);
            transform: translateY(8px);
  }
  80% {
    -webkit-transform: translateY(6.4px);
            transform: translateY(6.4px);
  }
  90% {
    -webkit-transform: translateY(-6.4px);
            transform: translateY(-6.4px);
  }
}
@keyframes shake-vertical {
  0%,
  100% {
    -webkit-transform: translateY(0);
            transform: translateY(0);
  }
  10%,
  30%,
  50%,
  70% {
    -webkit-transform: translateY(-8px);
            transform: translateY(-8px);
  }
  20%,
  40%,
  60% {
    -webkit-transform: translateY(8px);
            transform: translateY(8px);
  }
  80% {
    -webkit-transform: translateY(6.4px);
            transform: translateY(6.4px);
  }
  90% {
    -webkit-transform: translateY(-6.4px);
            transform: translateY(-6.4px);
  }
}
