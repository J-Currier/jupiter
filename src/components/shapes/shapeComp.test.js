import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { determineCorners, draw, checkBounds } from "./shapeComp";
import Shape from "./shapeComp";

describe("determine corners", () => {
  test("circle 0", () => {
    expect(determineCorners(1000, 1000, [200], 0, "circle")).toEqual([
      [1000, 1000]
    ]);
  });
  test("square 1", () => {
    expect(determineCorners(1000, 1000, [200], 1, "square")).toEqual([
      [1000, 1000],
      [1200, 1000],
      [1200, 800],
      [1000, 800]
    ]);
  });
  test("square 2", () => {
    expect(determineCorners(1000, 1000, [200], 2, "square")).toEqual([
      [1000, 1000],
      [1200, 1000],
      [1200, 1200],
      [1000, 1200]
    ]);
  });
  test("square 3", () => {
    expect(determineCorners(1000, 1000, [200], 3, "square")).toEqual([
      [1000, 1000],
      [800, 1000],
      [800, 1200],
      [1000, 1200]
    ]);
  });
  test("square 4", () => {
    expect(determineCorners(1000, 1000, [200], 4, "square")).toEqual([
      [1000, 1000],
      [800, 1000],
      [800, 800],
      [1000, 800]
    ]);
  });
  test("triangle 1", () => {
    expect(determineCorners(1000, 1000, [200, 300], 1, "triangle")).toEqual([
      [1000, 1000],
      [1200, 1000],
      [1000, 700]
    ]);
  });
  test("triangle 2", () => {
    expect(determineCorners(1000, 1000, [200, 300], 2, "triangle")).toEqual([
      [1000, 1000],
      [1200, 1000],
      [1000, 1300]
    ]);
  });
  test("triangle 3", () => {
    expect(determineCorners(1000, 1000, [200, 300], 3, "triangle")).toEqual([
      [1000, 1000],
      [800, 1000],
      [1000, 1300]
    ]);
  });
  test("triangle 4", () => {
    expect(determineCorners(1000, 1000, [200, 300], 4, "triangle")).toEqual([
      [1000, 1000],
      [800, 1000],
      [1000, 700]
    ]);
  });
  test("triangle 5", () => {
    expect(determineCorners(1000, 1000, [200, 300], 5, "triangle")).toEqual([
      [1000, 1000],
      [1300, 1000],
      [1000, 1200]
    ]);
  });
  test("triangle 6", () => {
    expect(determineCorners(1000, 1000, [200, 300], 6, "triangle")).toEqual([
      [1000, 1000],
      [700, 1000],
      [1000, 800]
    ]);
  });
  test("star 1", () => {
    expect(determineCorners(1000, 1000, [200], 1, "star")).toEqual([
      [1000, 1000],
      [1175, 875],
      [1375, 1000],
      [1300, 800],
      [1425, 650],
      [1250, 650],
      [1175, 475],
      [1100, 650],
      [925, 650],
      [1050, 800]
    ]);
  });
  test("star 2", () => {
    expect(determineCorners(1000, 1000, [200], 2, "star")).toEqual([
      [1000, 1000],
      [1125, 1175],
      [1000, 1375],
      [1200, 1300],
      [1350, 1425],
      [1350, 1250],
      [1525, 1175],
      [1350, 1100],
      [1350, 925],
      [1200, 1050]
    ]);
  });
  test("star 3", () => {
    expect(determineCorners(1000, 1000, [200], 3, "star")).toEqual([
      [1000, 1000],
      [825, 1125],
      [625, 1000],
      [700, 1200],
      [575, 1350],
      [750, 1350],
      [825, 1525],
      [900, 1350],
      [1075, 1350],
      [950, 1200]
    ]);
  });
  test("star 4", () => {
    expect(determineCorners(1000, 1000, [200], 4, "star")).toEqual([
      [1000, 1000],
      [875, 825],
      [1000, 625],
      [800, 700],
      [650, 575],
      [650, 750],
      [475, 825],
      [650, 900],
      [650, 1075],
      [800, 950]
    ]);
  });
});
describe("checkbounds", () => {
  test("square y out", () => {
    let points = [
        [0, 2001],
        [2000, 2001],
        [2000, 2000],
        [0, 2000]
      ],
      width = 2000,
      height = 2000,
      horizontalCall = jest.fn(),
      verticalCall = jest.fn();
    checkBounds(points, width, height, horizontalCall, verticalCall);
    expect(horizontalCall.mock.calls.length).toBe(0);
    expect(verticalCall.mock.calls.length).toBe(1);
  });
  test("square x out", () => {
    let points = [
        [-1, 0],
        [2000, 0],
        [2000, 2000],
        [-1, 2000]
      ],
      width = 2000,
      height = 2000,
      horizontalCall = jest.fn(),
      verticalCall = jest.fn();
    checkBounds(points, width, height, horizontalCall, verticalCall);
    expect(horizontalCall.mock.calls.length).toBe(1);
    expect(verticalCall.mock.calls.length).toBe(0);
  });
  test("circle y out", () => {
    let points = [[1000, 1900]],
      width = 2000,
      height = 2000,
      horizontalCall = jest.fn(),
      verticalCall = jest.fn(),
      radius = 200;
    checkBounds(points, width, height, horizontalCall, verticalCall, radius);
    expect(horizontalCall.mock.calls.length).toBe(0);
    expect(verticalCall.mock.calls.length).toBe(1);
  });
  test("circle out", () => {
    let points = [[-100, 1000]],
      width = 2000,
      height = 2000,
      horizontalCall = jest.fn(),
      verticalCall = jest.fn(),
      radius = 200;
    checkBounds(points, width, height, horizontalCall, verticalCall, radius);
    expect(horizontalCall.mock.calls.length).toBe(1);
    expect(verticalCall.mock.calls.length).toBe(0);
  });
});
describe("container", () => {
  let container = null;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
  test("draw", () => {
    let id = 1,
      width = 2000,
      height = 2000,
      shape = "square",
      corners = [
        [1000, 1000],
        [1200, 1000],
        [1200, 800],
        [1000, 800]
      ];
    const canvas = document.createElement("canvas");
    canvas.setAttribute("id", id);
    container.appendChild(canvas);
    const context = draw(id, width, height, shape, corners);
    const events = context.__getEvents();
    const path = context.__getPath();
    const drawCalls = context.__getDrawCalls();
    expect(events).toMatchSnapshot();
    expect(path).toMatchSnapshot();
    expect(drawCalls).toMatchSnapshot();
  });
  test("render", () => {
    const moveBack_shakeHorizontal = jest.fn();
    const moveBack_shakeVertical = jest.fn();
    let shapeInfo = {
      id: 1,
      position: [0, 0, [100], 1],
      fillColour: "rgba(255, 77, 0, 1)",
      borderColour: "rgba(255, 255, 255, 1)",
      borderWidth: 10,
      shapeClassName: "square"
    };
    act(() => {
      render(
        <Shape
          shape={"square"}
          shapeInfo={shapeInfo}
          moveBack_shakeHorizontal={moveBack_shakeHorizontal}
          moveBack_shakeVertical={moveBack_shakeVertical}
        />,
        container
      );
    });
  });
});
