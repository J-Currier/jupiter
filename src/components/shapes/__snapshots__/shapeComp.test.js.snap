// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container draw 1`] = `
Array [
  Object {
    "props": Object {
      "height": 2000,
      "width": 2000,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 1000,
      "y": 1000,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  Object {
    "props": Object {
      "x": 1200,
      "y": 1000,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "x": 1200,
      "y": 800,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "x": 1000,
      "y": 800,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "x": 1000,
            "y": 1000,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        Object {
          "props": Object {
            "x": 1200,
            "y": 1000,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 1200,
            "y": 800,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 1000,
            "y": 800,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "x": 1000,
            "y": 1000,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        Object {
          "props": Object {
            "x": 1200,
            "y": 1000,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 1200,
            "y": 800,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 1000,
            "y": 800,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;

exports[`container draw 2`] = `
Array [
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 1000,
      "y": 1000,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  Object {
    "props": Object {
      "x": 1200,
      "y": 1000,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "x": 1200,
      "y": 800,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "x": 1000,
      "y": 800,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
]
`;

exports[`container draw 3`] = `
Array [
  Object {
    "props": Object {
      "height": 2000,
      "width": 2000,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "x": 1000,
            "y": 1000,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        Object {
          "props": Object {
            "x": 1200,
            "y": 1000,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 1200,
            "y": 800,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 1000,
            "y": 800,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "x": 1000,
            "y": 1000,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        Object {
          "props": Object {
            "x": 1200,
            "y": 1000,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 1200,
            "y": 800,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 1000,
            "y": 800,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;
